application/src/ext/java/org/opentripplanner/ext/fares/impl/gtfs/DistanceMatcher.java

```diff
@@ -0,0 +1,24 @@
+package org.opentripplanner.ext.fares.impl.gtfs;
+
+import org.opentripplanner.ext.fares.model.FareDistance;
+import org.opentripplanner.ext.fares.model.FareLegRule;
+import org.opentripplanner.model.plan.TransitLeg;
+import org.opentripplanner.transit.model.basic.Distance;
+
+public class DistanceMatcher {
+
+  static boolean matchesDistance(TransitLeg leg, FareLegRule rule) {
+    // If no valid distance type is given, do not consider distances in fare computation
+    FareDistance distance = rule.fareDistance();
+    if (distance instanceof FareDistance.Stops(int min, int max)) {
+      var numStops = leg.listIntermediateStops().size();
+      return numStops >= min && max >= numStops;
```

Comment:
I think the max distance (the number of stops in this case) should be excluded per https://github.com/google/transit/blob/123a2907b5943605cb7dce41f4691b08ce3b8d44/gtfs/spec/en/reference.md?plain=1#L513, or if this is not related to the fares v2 specification, add a comment explaining why.